### DISCUSSION TITLE
managed to get python interface running partially, but fail to resolve HDF5 version mismatch

### DISCUSSION THREAD

**Discussion by g3grau** [thliebig/openEMS-Project#272] [Q&A Usage]
Hi,

I somehow managed to get the python interface running (only within a conda env), but I realize that this as well as the matlab/octave version crash on any antenna related example. The most likely reason is this version mismatch:

```
               hdf5   -- Version: 1.12.3
                          compiled against: HDF5 library version: 1.14.3
```

AFAIK, I initially only had 1.12.3 installed (from openSuse Tumbleweed Science repo), so I don't know where the 1.14.3 originally comes from. Meanwhile I installed HDF5 1.14.3 from source and set LD_LIBRARY_PATH, PATH and HDF5_ROOT to the respective places of it.
It didn't change anything. I recompiled/reinstalled CSXCAD and openEMS with these settings, still the same issue. I uninstalled and reinstalled hyp5 both with conda and pip, no change. 
I'm clueless how to resolve this as everything *should* point to the 1.14.3, but apparently still the older one in /usr/lib64 seems to be used. Any change this is hardcoded anywhere?

BTW, on the way through the installation jungle I had to 
```
cd /usr/lib4
ln -s ln -s libCSXCAD.so.0.6.3 libCSXCAD.so
```
to make things work. In parallel to setting LD_LIBRARY_PATH I renamed the /usr/lib64/ libCSXCAD.so and libhdf5.so to *.inactive ...that doesn't do any change, so the OS versions are likely not used

**Comment by thliebig** [comment]
I am only aware of this problem on Windows soo far. The issue is that openEMS python uses/imports the h5py module and is build against hdf5.
Thus if h5py is imported first CSXCAD may find the wrong *.so  and vice versa.
But usually this is not an issue on linux as everything is build against the same hdf5.so.
Maybe this is an issue with conda? Why not use the "vanilla" python?
